wgsl/index.bs

```diff
@@ -2703,6 +2672,83 @@ Issue: (dneto): Bitwise-complement is under discussion.  https://github.com/gpuw
   <tr><td>*e1* : *T*<br>*e2* : *T*<br>*T* is *BoolVec*<td>`e1 & e2` : *T*<td>Component-wise logical "and"
 </table>
 
+# Built-in functions # {#builtin-functions}
+
+Certain functions are always available in a [SHORTNAME] program,
+and are provided by the implementation.
+These are called *built-in functions*.
+
+Since a built-in function is always in scope, it is an error to attempt to redefine
+one or to use the name of a built-in function as an identifier for any other
+kind of declaration.
+
+Unlike ordinary functions defined in a [SHORTNAME] program,
+a built-in function may use the same function name with different
+sets of parameters.
+In other words, a built-in function may have more than one *overload*,
+but ordinary function definitions in [SHORTNAME] may not.
+
+When calling a built-in function, all arguments to the function are evaluated
+before function evaulation begins.
+
+TODO(dneto): Elaborate the descriptions of the built-in functions.  So far I've only reorganized
+the contents of the existing table.
+
+## Logical built-in functions ## {#logical-builtin-functions}
+
+<table class='data'>
+  <thead>
+    <tr><td>Logical built-in functions<td>SPIR-V
+  </thead>
+  <tr><td>all(BoolVec) -&gt; bool<td>OpAll
+  <tr><td>any(BoolVec) -&gt; bool<td>OpAny
+  <tr><td>select(*T*,*T*,bool) -&gt; *T*<td>
+       For scalar or vector type *T*.
+       `select(a,b,c)` evaluates to *a* when *c* is true, and *b* otherwise.
+  <tr><td>select(vec*N*&lt;*T*&gt;,vec*N*&lt;*T*&gt;,vec*N*&lt;bool&gt;) -&gt; vec*N*&lt;*T*&gt;<td>
+       For scalar type *T*.
+       `select(a,b,c)` evaluates to a vector with component *i* being `select(a[i], b[i], c[i])`.
+</table>
+
+## Value-testing built-in functions ## {#value-testing-builtin-functions}
+
+<table class='data'>
+  <thead>
+    <tr><td>Logical built-in functions<td>SPIR-V
+  </thead>
+  <tr><td>is_finite(float) -&gt; bool<td>OpIsFinite
+  <tr><td>is_inf(float) -&gt; bool<td>OpIsInf
+  <tr><td>is_nan(float) -&gt; bool<td>OpIsNan
+  <tr><td>is_normal(float) -&gt; bool<td>OpIsNormal
+</table>
+
+## Vector built-in functions ## {#vector-builtin-functions}
+
+<table class='data'>
+  <thead>
+    <tr><td>Built-in<td>SPIR-V
```

Comment:
should this be "Data built-in"?

Reply:
I'll go with "Vector built-in functions" to match the section heading.